carla/models/catalog/catalog.py

```diff
@@ -48,17 +48,19 @@ def __init__(
         use_pipeline : bool, optional
             If true, the model uses a pipeline before predict and predict_proba to preprocess the input data.
         """
-        super().__init__(data)
         self._backend = backend
 
         if self._backend == "pytorch":
             ext = "pt"
+            encoding_method = "OneHot"
         elif self._backend == "tensorflow":
             ext = "h5"
+            encoding_method = "OneHot_drop_binary"
         else:
             raise ValueError(
                 "Backend not available, please choose between pytorch and tensorflow"
             )
+        super().__init__(data, encoding_method=encoding_method)
```

Comment:
In the MLModelCatalog there is now no option to set the encoding method by hand. Does it really only depend on the TF/PyTorch backend?

Reply:
This implementation only concerns our pre-trained models in the catalog, and currently, it is the case that the encoding depends on the backend. Giving the user the choice of encoding for our catalog would cause failures in various recourse methods. If a user has his own black-box model, he could use every encoding he wants by inheriting the MLModel class.